File: app-code/cz-acme/simple-validation-java/src/main/java/com/acme/simple/validation/controller/api/v1/UserController.java
package com.acme.simple.validation.controller.api.v1;

import com.themusketeers.sbnative.domain.User;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * User API Controller.
 * <p><b>Path:</b>{@code api/v1/users}</p>
 *
 * @author COQ - Carlos Adolfo Ortiz Q.
 */
@RestController
@RequestMapping("api/v1/users")
public record UserController() {

    @PostMapping
    public User insertUser(@Valid @RequestBody User user) {
        return new User(user.id(), user.name() + " NEWST", user.address() + " NEWST");
    }


}

File: app-code/cz-acme/simple-validation-kotlin/src/main/kotlin/com/acme/simple/validation/controller/api/v1/UserController.kt
package com.acme.simple.validation.controller.api.v1

import com.acme.simple.validation.domain.User
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("api/v1/users")
class UserController {
    @PostMapping
    fun insertUser(@RequestBody @Valid user:  User): User {
        return User(user.id, user.name + " NEWST", user.address + " NEWST")
    }
}
